### DISCUSSION TITLE
Multiple code locations for agent issue

### DISCUSSION THREAD

**Discussion by spartakbughdaryan** [dagster-io/dagster#19165] [Q&A] [answered]
Hello,
I'm currently managing a setup where I have two separate agents deployed to ECS, corresponding to two code locations: staging and production, with each agent intended for each environment respectively. However, I'm encountering an issue where each agent attempts to run on both of the code locations, which is not the intended behavior.
I have enabled the isolated_agents configuration, hoping it would resolve this issue, but it doesn't seem to be effective. In my attempt to troubleshoot, I came across two different sets of documentation regarding where the isolated_agents configuration should be placed. One document suggests that it should be included in the dagster.yaml file, while the other indicates that it should be within the dagster.yaml file but inside the config section.
I'm seeking clarity on this matter and any assistance on how to correctly configure the agents so that each one operates solely within its designated environment (staging or production) would be greatly appreciated. Here are the sources I've been referring to:
[Running Multiple Agents](https://docs.dagster.io/dagster-cloud/deployment/agents/running-multiple-agents)
[Amazon ECS Configuration Reference](https://docs.dagster.io/dagster-cloud/deployment/agents/amazon-ecs/configuration-reference#per-deployment-configuration)
Any help or guidance on this issue would be highly valuable. Thank you in advance!

**Comment by shalabhc** [comment, marked as answer]
Hi @spartakbughdaryan - if you want to route specific code locations to specific agents, this is a feature that was just released today in [1.6.0](https://github.com/dagster-io/dagster/releases/tag/1.6.0) called agent routing: https://master.dagster.dagster-docs.io/dagster-cloud/deployment/agents/running-multiple-agents#routing-requests-to-specific-agents. Prior to this, code location based routing was not available and the use case for multiple agents was only for redundancy, so all agents handled all code locations within the deployment. EDIT: note you will need to be on [Dagster Cloud Pro](https://dagster.io/pricing) to use agent routing.

Also note if you're on Dagster Cloud Pro, you can have more than one deployment. Often 'production' and 'staging' are implemented as different deployments and it allows you to the same set of code locations within each.

**Reply by spartakbughdaryan** [reply]
Thanks for the respond. Got it. Can you please tell me what is the functionality of the `isolated_agents` argument in dagster.yaml file?
